File: RMA-backend-java/src/main/java/at/htlpinkafeld/RMA_backend_java/pojo/Event.java
package at.htlpinkafeld.RMA_backend_java.pojo;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.sql.Date;

public class Event implements Identifiable {
    private int id;
    private String name;
    private Date startDate;
    private Date endDate;

    @JsonCreator
    public Event(@JsonProperty("id") String id, @JsonProperty("name") String name,
                 @JsonProperty("startDate") Date startDate, @JsonProperty("endDate") Date endDate){
        if(id == null){
            this.init(-1,name,startDate,endDate);
        } else {
            this.init(Integer.parseInt(id),name,startDate,endDate);
        }
    }

    public Event(int id, String name,Date startDate, Date endDate) {
        this.init(id,name,startDate,endDate);
    }
    private void init(int id, String name, Date startDate, Date endDate){
        this.id = id;
        this.name = name;
        this.startDate = startDate;
        this.endDate = endDate;
    }

    public Event(String name, Date startDate, Date endDate) {
        this.init(-1,name,startDate,endDate);
    }

    public int getID() {
        return id;
    }

    public void setID(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Date getStartDate() {
        return startDate;
    }

    public void setStartDate(Date startDate) {
        this.startDate = startDate;
    }

    public Date getEndDate() {
        return endDate;
    }

    public void setEndDate(Date endDate) {
        this.endDate = endDate;
    }
}

File: Frontend/angular-rma/src/Event.ts
export class Event{

    constructor(private name: string, private startDate: Date, 
        private endDate?: Date, private id?: number){

    }

    get ID(){
        return this.id;
    }

    set ID(id: number){
        this.id = id;
    }

    get Name(){
        return this.name;
    }

    set Name(name: string){
        this.name = name;
    }

    get StartDate(){
        return this.startDate;
    }

    set StartDate(startDate: Date){
        this.startDate = startDate;
    }

    get EndDate(){
        return this.endDate;
    }

    set EndDate(endDate: Date){
        this.endDate = endDate;
    }

    public static eventOfObject(object: any): Event{
        return new Event(object.name, object.startDate, object.endDate, object.id);
    }

}